File: aritma_task.Server/Models/Loan.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace aritma_task.Server.Models;

public class Loan
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    
    public int LoanTypeId { get; set; }

    [ForeignKey("LoanTypeId")]
    public virtual LoanType LoanType { get; set; }

    public decimal Amount { get; set; }

    public int TermInYears { get; set; }

    public DateTime StartDate { get; set; }
}

File: aritma_task.client/src/models/Loan.ts
import LoanType from "./LoanType.ts";
import {AmortizationDetail} from "./AmortizationDetail.ts";

export default interface Loan {
    id: number;
    loanType: LoanType;
    amount: number;
    termInYears: number;
    amortizationSchedule: AmortizationDetail[];
}